python/packages/ai/teams/ai/prompts/default_prompt_manager.py

```diff
@@ -44,13 +49,13 @@ def add_function(
         :param allow_overrides: Whether to allow overriding an existing function with the same name.
         """
         if not allow_overrides and self._functions.get(name):
-            raise Exception(f"Function {name} already exists")
+            raise AIException(f"Function {name} already exists")
 
         self._functions[name] = handler
         return self
 
     async def render_prompt(
-        self, context: TurnContext, state: TurnState, name_or_template: str | PromptTemplate
+        self, context: TurnContext, state: TurnState, name_or_template: Union[str, PromptTemplate]
```

Comment:
I'm fine with this change. But I remember `poetry run fmt` changes this to `str | PromptTemplate` for me. Just FYI. I will look at this further.

Reply:
ya I had to change this cause the `|` syntax was failing to compile